### PR TITLE
Factor pwm, psol and buzzer together

### PR DESCRIPTION
**Pull request by asmodai27**
Build a generic pwm driven actuator for psol and blower
Buzzer is not really an actuator and stays a member of hal for now

# Description
We basically had 3 modules that implemented the same functionalities of outputting a pwm signal to a GPIO pin. I factored those in together, and made it so we can instantiate and initialize those separately from `hal`.
This also addresses several todos and makes the instantiation of PWM driven actuators agnostic to hal.
A this point, we can probably create unit tests for actuators class, with a little mocking effort on the PWM and/or pinch valve side.

Please note:
- One caveat of initializing pwm pins outside of hal is that if we never instantiate them (for example in an integration test), the pins will have a high output, making the actuators behave at their max setting.
- I could not fully test psol output, as I don't have the hardware for it.

# General checklist:

- [x] I have performed a self-review, including - looked through the `Files changed` tab, browsed repository in my branch
- [ ] I have made corresponding changes to the documentation - to reflect changes in code, electrical or mechanical design
- [ ] All new documentation or graphics follow the [documentation style guide](https://github.com/RespiraWorks/Ventilator/wiki/Documentation-style-guide)
- [ ] All new content is linked, easily discoverable, does not require too many clicks
- [x] I have reviewed any other relevant parts of our [contributor wiki](https://github.com/RespiraWorks/Ventilator/wiki) to ensure that this PR conforms to the standards
- [x] I have given the PR a descriptive name (prefixed with **WIP** if it is not yet ready for review)
- [x] I have tagged relevant reviewers

## Code checklist:

- [x] All new code follows our [code style](https://github.com/RespiraWorks/Ventilator/wiki/Code-style)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have described tests I used to verify my changes, instructions for reproducing them and any relevant configuration details.
- [x] I ran the unit test suite and verified that all tests pass - new and old, locally and on CI
- [x] I performed a clean build and verified that there were no warnings
- [x] I ran our static analysis tools and verified there were no warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] All source files have license headers

### CODE REVIEW

**Comment by asmodai27** [comment]
Remaining comment addressed separately, in #1191

File: software/controller/lib/hal/hal_mock.cpp [hunk 1]
```diff
@@ -15,8 +15,11 @@ limitations under the License.
 
 #if !defined(BARE_STM32)
 
+#include <cstring>
+
 #include "hal.h"
 
+Frequency HalApi::GetCpuFreq() { return hertz(0); }
```
Comment (martukas) [inline software/controller/lib/hal/hal_mock.cpp:22]: Is this a good assumption for a mock HAL? If not yet sure, let's leave a TODO?

File: software/controller/lib/hal/system_timer.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@ void SystemTimer::initialize(uint32_t cpu_frequency_MHz) {
   // -1 until the clock wraps back to zero and generates an interrupt. This
   // setting will cause an interrupt every 10,000 clocks or 1 millisecond
   tmr->auto_reload = 9999;
-  tmr->prescaler = (cpu_frequency_MHz / 10 - 1);
+  tmr->prescaler = static_cast<uint32_t>(cpu_frequency.megahertz() / 10.0f - 1.0f);
```
Comment (martukas) [inline software/controller/lib/hal/system_timer.cpp:45]: Should this be an explicit rounding, floor or ceiling? Not clear which is intended. May be a few other instances of something like this that I missed.

**Review by martukas — APPROVED** [review]
So this should be the last set of changes I'll ask for. Mostly small things (such as leaving some TODOs for future improvements), except for splitting off PwmActuator away from the abstract class header. I closed most old comments as "resolved" but added follow-up comments on some, so revisit them if you can.

So I'm filing it as approve and I trust that you will clear it all up as you best see fit.

**Review by asmodai27 — COMMENTED** [review]
Thanks for the review.

**Pull request merged**